spark/v4.0/spark/src/main/java/org/apache/iceberg/spark/SparkTableUtil.java

```diff
@@ -875,7 +875,13 @@ private static void deleteManifests(FileIO io, List<ManifestFile> manifests) {
   public static Dataset<Row> loadTable(SparkSession spark, Table table, long snapshotId) {
     SparkTable sparkTable = new SparkTable(table, snapshotId, false);
     DataSourceV2Relation relation = createRelation(sparkTable, ImmutableMap.of());
-    return Dataset.ofRows(spark, relation);
+    if (!(spark instanceof org.apache.spark.sql.classic.SparkSession)) {
+      throw new IllegalArgumentException(
+          "spark is supposed to be org.apache.spark.sql.classic.SparkSession");
```

Comment:
Wouldn't this be something like "Cannot use loadTable with Spark Connect" ?

Reply:
We probably don't need to take consideration of Spark Connect? I guess we don't need to run Iceberg on client side?